### ISSUE TITLE
t2204-job-info: fails under valgrind

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#2665]
When run with
```
$ FLUX_TEST_VALGRIND=t ./t2204-job-info.t -d -v
```
Two tests always fail.

```
expecting success: 
        flux job list -a > list_reload.out &&
        for count in `seq 1 16`; do \
            echo "32" >> list_reload_state.exp; \
        done &&
        cat list_reload.out | jq .state  > list_reload_state.out &&
        test_cmp list_reload_state.exp list_reload_state.out &&
        tac job_ids_running.out >> list_reload_ids.exp &&
        tac job_ids_pending.out >> list_reload_ids.exp &&
        cat job_ids_inactive.out >> list_reload_ids.exp &&
        cat list_reload.out | jq .id > list_reload_ids.out &&
        test_cmp list_reload_ids.exp list_reload_ids.out

--- list_reload_state.exp       2020-01-22 17:09:10.700899000 +0000
+++ list_reload_state.out       2020-01-22 17:09:10.704895000 +0000
@@ -1,4 +1,4 @@
-32
+16
 32
 32
 32
not ok 20 - job-info: list successfully reconstructed
#       
#               flux job list -a > list_reload.out &&
#               for count in `seq 1 16`; do \
#                   echo "32" >> list_reload_state.exp; \
#               done &&
#               cat list_reload.out | jq .state  > list_reload_state.out &&
#               test_cmp list_reload_state.exp list_reload_state.out &&
#               tac job_ids_running.out >> list_reload_ids.exp &&
#               tac job_ids_pending.out >> list_reload_ids.exp &&
#               cat job_ids_inactive.out >> list_reload_ids.exp &&
#               cat list_reload.out | jq .id > list_reload_ids.out &&
#               test_cmp list_reload_ids.exp list_reload_ids.out
#       

expecting success: 
        flux job stats | jq -e ".job_states.depend == 0" &&
        flux job stats | jq -e ".job_states.sched == 0" &&
        flux job stats | jq -e ".job_states.run == 0" &&
        flux job stats | jq -e ".job_states.cleanup == 0" &&
        flux job stats | jq -e ".job_states.inactive == 16" &&
        flux job stats | jq -e ".job_states.total == 16"

true
true
true
false
not ok 21 - job stats lists jobs in correct state (all inactive)
#       
#               flux job stats | jq -e ".job_states.depend == 0" &&
#               flux job stats | jq -e ".job_states.sched == 0" &&
#               flux job stats | jq -e ".job_states.run == 0" &&
#               flux job stats | jq -e ".job_states.cleanup == 0" &&
#               flux job stats | jq -e ".job_states.inactive == 16" &&
#               flux job stats | jq -e ".job_states.total == 16"
#       
```

**Comment by dongahn** [comment]
OK. I think this was one of the problems I saw from my PR. https://github.com/flux-framework/flux-core/pull/2654#issuecomment-576538255. There were other tests which were failing but one at a time.

**Comment by chu11** [comment]
This seems to be an eventual consistency issue, where on re-load, one job just hasn't quite completed re-loading.  the state is `16` indicating it's in the cleanup state instead of inactive (even though it was canceled in an earlier test).

```
>cat trash-directory.t2204-job-info/list_reload.out
{"id": 370289934336, "userid": 8556, "priority": 0, "t_submit": 1579718893.492768, "state": 16, "name": "hostname", "ntasks": 1, "t_depend": 1579718893.492768, "t_sched": 1579718893.7186728, "t_run": 0.0, "t_cleanup": 1579718895.4583695, "t_inactive": 0.0}
```

I'll put some racyness checks in to solve some of these issues.  However, some I believe will have to be solved when we fix #2592.
